yellowbrick/contrib/missing/bar.py

```diff
@@ -95,31 +93,28 @@ def __init__(self, width=0.5, color='black', colors=None, classes=None, **kwargs
 
         self.colors = color_palette(kwargs.pop('colors', None), n_colors)
 
-    def get_nan_col_counts(self, **kwargs):
-        # where matrix contains strings, handle them
-        if np.issubdtype(self.X.dtype, np.string_) or np.issubdtype(self.X.dtype, np.unicode_):
-            mask = np.where( self.X == '' )
-            nan_matrix = np.zeros(self.X.shape)
-            nan_matrix[mask] = np.nan
+        # set later
+        self.nan_col_counts_ = []
 
-        else:
-            nan_matrix = self.X.astype(np.float)
+    def get_nan_col_counts(self, X, y, **kwargs):
+        """Calculate the number of nan values on a per column basis.
+        """
+        nan_matrix = self.create_nan_matrix(X)
 
-        if self.y is None:
-            nan_col_counts = [np.count_nonzero(np.isnan(col)) for col in nan_matrix.T]
-            return nan_col_counts
+        if y is None:
+            self.nan_col_counts_ = [np.count_nonzero(np.isnan(col)) for col in nan_matrix.T]
+            return self.nan_col_counts_
 
         else:
             # add in counting of np.nan per target y by column
-            nan_counts = []
-            for target_value in np.unique(self.y):
+            for target_value in np.unique(y):
```

Comment:
Thank you for removing the data storage on the class! I think this is going to make the missing values visualizer much more performant!